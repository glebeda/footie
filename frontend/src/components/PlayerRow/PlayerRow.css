.cancel-icon-cell {
  width: 1%;
  white-space: nowrap;
}

.row-swiping {
    transform: translateX(10px);
    transition: transform 0.3s ease;
}
