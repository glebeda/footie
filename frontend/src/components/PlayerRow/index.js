import React, { forwardRef } from 'react';
import PaidCheckbox from '../PaidCheckbox';
import CancelIcon from '@mui/icons-material/Cancel'
import './PlayerRow.css'
import { TableCell, TableRow, IconButton } from '@mui/material'

const PlayerRow = forwardRef(({ player, index, handleOpenDialog, isHighlighting, isRemoving, showAlert, hideAlert, isSubstitute }, ref) => {

  const rowClasses = [
    isSubstitute ? 'substitute-row' : '', 
    isHighlighting ? 'highlighted-row' : '', 
    isRemoving ? 'removing' : '',
  ].filter(Boolean).join(' ');

  return (
    <TableRow ref={ref} className={rowClasses}>
      <TableCell component='th' scope='row'>
        {index}
      </TableCell>
      <TableCell>{player.name}</TableCell>
      <TableCell align='right'>
        <PaidCheckbox 
          player={player} 
          disabled={isSubstitute} 
          showAlert={showAlert} 
          hideAlert={hideAlert} 
        />
      </TableCell>
      <TableCell align='right' className='cancel-icon-cell'>
        <IconButton onClick={() => handleOpenDialog(player)}>
          <CancelIcon />
        </IconButton>
      </TableCell>
    </TableRow>
  );
});

export default PlayerRow
