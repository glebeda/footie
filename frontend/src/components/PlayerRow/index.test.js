import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import PlayerRow from './index';

jest.mock('../PaidCheckbox', () => () => <div>PaidCheckbox Mock</div>);
jest.mock('@mui/icons-material/Cancel', () => () => <span>CancelIcon Mock</span>);

describe('PlayerRow', () => {
  const playerMock = { name: 'Player 1', paid: false, PlayerId: '1', gameId: 'game1' };
  const handleOpenDialogMock = jest.fn();
  const showAlertMock = jest.fn();
  const hideAlertMock = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <PlayerRow
        player={playerMock}
        index={0}
        handleOpenDialog={handleOpenDialogMock}
        highlightedIndex={null}
        isRemoving={false}
        showAlert={showAlertMock}
        hideAlert={hideAlertMock}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('calls handleOpenDialog on cancel icon click', () => {
    render(
      <PlayerRow
        player={playerMock}
        index={0}
        handleOpenDialog={handleOpenDialogMock}
        highlightedIndex={null}
        isRemoving={false}
        showAlert={showAlertMock}
        hideAlert={hideAlertMock}
      />
    );

    fireEvent.click(screen.getByText('CancelIcon Mock'));
    expect(handleOpenDialogMock).toHaveBeenCalledWith(playerMock);
  });

  it('applies highlighted-row class when index equals highlightedIndex', () => {
    render(
      <PlayerRow
        player={playerMock}
        index={0}
        handleOpenDialog={handleOpenDialogMock}
        highlightedIndex={0}
        isRemoving={false}
      />
    );

    expect(screen.getByText(playerMock.name).closest('tr')).toHaveClass('highlighted-row');
  });

  it('does not apply highlighted-row class when index does not equal highlightedIndex', () => {
    render(
      <PlayerRow
        player={playerMock}
        index={0}
        handleOpenDialog={handleOpenDialogMock}
        highlightedIndex={1} // Different index
        isRemoving={false}
      />
    );

    expect(screen.getByText(playerMock.name).closest('tr')).not.toHaveClass('highlighted-row');
  });

  it('applies removing class when isRemoving is true', () => {
    render(
      <PlayerRow
        player={playerMock}
        index={0}
        handleOpenDialog={handleOpenDialogMock}
        highlightedIndex={null}
        isRemoving={true}
      />
    );

    expect(screen.getByText(playerMock.name).closest('tr')).toHaveClass('removing');
  });

  it('does not apply removing class when isRemoving is false', () => {
    render(
      <PlayerRow
        player={playerMock}
        index={0}
        handleOpenDialog={handleOpenDialogMock}
        highlightedIndex={null}
        isRemoving={false}
      />
    );

    expect(screen.getByText(playerMock.name).closest('tr')).not.toHaveClass('removing');
  });

  it('assigns ref to the TableRow component', () => {
    const ref = React.createRef();
    render(
      <table>
        <tbody>
          <PlayerRow
            ref={ref}
            player={playerMock}
            index={0}
            handleOpenDialog={handleOpenDialogMock}
            highlightedIndex={0}
            isRemoving={false}
            showAlert={showAlertMock}
            hideAlert={hideAlertMock}
          />
        </tbody>
      </table>
    );
  
    expect(ref.current.tagName).toBe('TR');
  });

});
