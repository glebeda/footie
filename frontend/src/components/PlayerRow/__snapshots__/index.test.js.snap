// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerRow renders correctly 1`] = `
<DocumentFragment>
  1Player 1
  <div>
    PaidCheckbox Mock
  </div>
  <button
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span>
      CancelIcon Mock
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
